NitroxServer/GameLogic/Bases/BaseData.cs

```diff
@@ -1,12 +1,14 @@
-﻿using NitroxModel.DataStructures.GameLogic;
+﻿using System;
+using NitroxModel.DataStructures.GameLogic;
 using System.Collections.Generic;
+using System.Linq;
 using Newtonsoft.Json;
 using ProtoBufNet;
 
 namespace NitroxServer.GameLogic.Bases
 {
     [ProtoContract, JsonObject(MemberSerialization.OptIn)]
-    public class BaseData
+    public sealed record BaseData
```

Comment:
Do you mean why `record` instead of `class`? It automatic implements equality checks based on values and not reference. So huge chunks of repeating code (like `Equals(object obj)` and `GetHashCode()`) can be avoided. Only if List/Dictionaries are in the `record` we have to override with `Equals(T other)`.